File: MODUL4/Soal4/PRAK404-2410817120013-OktaviaLailaQodari.py
while True:
    print("\nPilih program")
    print("1. Penjumlahan")
    print("2. Pengurangan")
    print("3. Perkalian")
    print("4. Pembagian")
    print("5. Exit")
        
    pilihan = input("Masukkan Pilihan : ")
        
    if pilihan == '5':
        print("Terimakasih, telah menggunakan kalkulator Oktavia")
        break
        
    if pilihan in ['1', '2', '3', '4']:
        try:
            nilai1 = float(input("Masukkan nilai pertama : "))
            nilai2 = float(input("Masukkan nilai kedua : "))
        except ValueError:
            print("Input tidak valid, silakan masukkan angka.")
            continue
            
        if pilihan == '1':
            hasil = nilai1 + nilai2
            operasi = "Penjumlahan"
        elif pilihan == '2':
            hasil = nilai1 - nilai2
            operasi = "Pengurangan"
        elif pilihan == '3':
            hasil = nilai1 * nilai2
            operasi = "Perkalian"
        elif pilihan == '4':
            if nilai2 == 0:
                print("Error: Pembagian dengan nol tidak diperbolehkan.")
                continue
            hasil = nilai1 / nilai2
            operasi = "Pembagian"

        print(f"Hasil {operasi} antara {nilai1:.2f} dan {nilai2:.2f} adalah {hasil:.2f}")
    else:
        print("Input anda salah, silahkan coba lagi")

File: MODUL4/Soal4/PRAK404-2410817120013-OktaviaLailaQodari.c
#include <stdio.h>

int main() {
    int pilihan;
    float nilai1, nilai2, hasil;

    while (1) {
        printf("Pilih program\n");
        printf("1. Penjumlahan\n");
        printf("2. Pengurangan\n");
        printf("3. Perkalian\n");
        printf("4. Pembagian\n");
        printf("5. Exit\n");
        printf("Masukkan Pilihan : ");
        scanf("%d", &pilihan);

        if (pilihan == 5) {
            printf("Terimakasih, telah menggunakan kalkulator Oktavia\n");
            break;
        }

        if (pilihan >= 1 && pilihan <= 4) {
            printf("Masukkan nilai pertama : ");
            scanf("%f", &nilai1);
            printf("Masukkan nilai kedua : ");
            scanf("%f", &nilai2);

            switch (pilihan) {
                case 1:
                    hasil = nilai1 + nilai2;
                    printf("Hasil Penjumlahan antara %.2f dengan %.2f adalah %.2f\n", nilai1, nilai2, hasil);
                    break;
                case 2:
                    hasil = nilai1 - nilai2;
                    printf("Hasil Pengurangan antara %.2f dengan %.2f adalah %.2f\n", nilai1, nilai2, hasil);
                    break;
                case 3:
                    hasil = nilai1 * nilai2;
                    printf("Hasil Perkalian antara %.2f dengan %.2f adalah %.2f\n", nilai1, nilai2, hasil);
                    break;
                case 4:
                    if (nilai2 != 0) {
                        hasil = nilai1 / nilai2;
                        printf("Hasil Pembagian antara %.2f dengan %.2f adalah %.2f\n", nilai1, nilai2, hasil);
                    } else {
                        printf("Error: Pembagian dengan nol tidak diperbolehkan.\n");
                    }
                    break;
            }
        } else {
            printf("Input anda salah, silahkan coba lagi\n");
        }
    }

    return 0;
}